### DISCUSSION TITLE
Issue with  multiple wan interfaces

### DISCUSSION THREAD

**Discussion by shripad-nesecure** [strongswan/strongswan#1902] [General]
Hi, Im having issue with strongswan on openwrt 21.02,

when having more than 1 wan interfaces, once ipsec is up, im not able to send the traffic from same interface, on tcpdump im seeing traffic is going via wan but response is not coming back.

once this ipsec is up, im not able to send any traffic with source as wan, but i can send any traffic from wwan

if with same config router is having only 1 config, its working without any issue
My ipsec.conf 
`version 2

conn test123-test123_c
  left=%any
  right=139.59.60.111
  leftfirewall=yes
  rightfirewall=yes
  ikelifetime=8h
  lifetime=8h
  margintime=9m
  keyingtries=3
  dpdaction=none
  dpddelay=30s
  dpdtimeout=90s
  leftauth=psk
  rightauth=psk
  rightsubnet=0.0.0.0/0
  auto=start
  leftsubnet=192.168.8.0/24
  aggressive=no
  forceencaps=no
  type=tunnel
  keyexchange=ikev1
  esp=aes128-sha1-modp1024
  ike=aes128-sha1-modp1024

	
conn bypass
	left=127.0.0.1
	leftsubnet=192.168.8.0/24
	rightsubnet=192.168.8.0/24
	auto=route
	type=passthrough`

ipsec statusall
`Status of IKE charon daemon (strongSwan 5.9.2, Linux 5.4.229, mips):
  uptime: 35 minutes, since Sep 11 16:41:00 2023
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 3
  loaded plugins: charon aes des sha2 sha1 md4 md5 random nonce x509 revocation constraints pubkey pkcs1 pkcs8 pgp pem openssl gmp xcbc hmac kernel-netlink socket-default stroke vici updown eap-identity eap-mschapv2 xauth-generic
Listening IP addresses:
  192.168.1.75
  fda3:30a0:f51e:0:21e:42ff:fe39:b7d6
  fd0e:a5ee:c61:0:21e:42ff:fe39:b7d6
  fd0e:a5ee:c61::395
  25.57.202.79
  192.168.8.1
  fd0e:a5ee:c61:4::1
  fd18:d3df:3cf0::1
Connections:
test123-test123_c:  %any...139.59.60.111  IKEv1
test123-test123_c:   local:  uses pre-shared key authentication
test123-test123_c:   remote: [139.59.60.111] uses pre-shared key authentication
test123-test123_c:   child:  192.168.8.0/24 === 0.0.0.0/0 TUNNEL
Security Associations (1 up, 0 connecting):
test123-test123_c[1]: ESTABLISHED 35 minutes ago, 192.168.1.75[192.168.1.75]...139.59.60.111[139.59.60.111]
test123-test123_c[1]: IKEv1 SPIs: 1982a1d43ff59e65_i* ba2d4548dda28875_r, pre-shared key reauthentication in 7 hours
test123-test123_c[1]: IKE proposal: AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
test123-test123_c{1}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: c524c185_i c9ba82da_o
test123-test123_c{1}:  AES_CBC_128/HMAC_SHA1_96/MODP_1024, 48506 bytes_i (657 pkts, 0s ago), 67154 bytes_o (483 pkts, 99s ago), rekeying in 7 hours
test123-test123_c{1}:   192.168.8.0/24 === 0.0.0.0/0
`
ip a

`1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:1e:42:39:b7:d5 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::21e:42ff:fe39:b7d5/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:1e:42:39:b7:d6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.75/24 brd 192.168.1.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fda3:30a0:f51e:0:21e:42ff:fe39:b7d6/64 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fd0e:a5ee:c61:0:21e:42ff:fe39:b7d6/64 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fd0e:a5ee:c61::395/128 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::21e:42ff:fe39:b7d6/64 scope link 
       valid_lft forever preferred_lft forever
4: ip6tnl0@NONE: <NOARP> mtu 1452 qdisc noop state DOWN group default qlen 1000
    link/tunnel6 :: brd ::
5: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
6: gre0@NONE: <NOARP> mtu 1476 qdisc noop state DOWN group default qlen 1000
    link/gre 0.0.0.0 brd 0.0.0.0
7: gretap0@NONE: <BROADCAST,MULTICAST> mtu 1462 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
8: erspan0@NONE: <BROADCAST,MULTICAST> mtu 1450 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
9: ip6gre0@NONE: <NOARP> mtu 1448 qdisc noop state DOWN group default qlen 1000
    link/gre6 :: brd ::
11: wwan0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN group default qlen 1000
    link/none 
    inet 25.57.202.79/32 brd 255.255.255.255 scope global wwan0
       valid_lft forever preferred_lft forever
    inet6 fe80::6fe2:8501:2b00:e861/64 scope link stable-privacy 
       valid_lft forever preferred_lft forever
12: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:1e:42:39:b7:d5 brd ff:ff:ff:ff:ff:ff
    inet 192.168.8.1/24 brd 192.168.8.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd0e:a5ee:c61:4::1/62 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fd18:d3df:3cf0::1/60 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::21e:42ff:fe39:b7d5/64 scope link 
       valid_lft forever preferred_lft forever
13: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP group default qlen 1000
    link/ether 00:1e:42:39:b7:d5 brd ff:ff:ff:ff:ff:ff
14: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP group default qlen 1000
    link/ether 00:1e:42:39:b7:d7 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::21e:42ff:fe39:b7d7/64 scope link 
       valid_lft forever preferred_lft forever

`

ip route

`default via 192.168.1.1 dev eth1 proto static src 192.168.1.75 metric 2 
default dev wwan0 proto static scope link src 25.57.202.79 metric 4 
25.57.202.79 dev wwan0 proto static scope link metric 4 
192.168.1.0/24 dev eth1 proto static scope link metric 2 
192.168.8.0/24 dev br-lan proto static scope link metric 1
`
ip route show table 220
`default via 192.168.1.1 dev eth1 proto static src 192.168.8.1 
`
ip xfrm policy
`src 192.168.8.0/24 dst 0.0.0.0/0 
	dir out priority 387711 
	tmpl src 192.168.1.75 dst 139.59.60.111
		proto esp spi 0xc9ba82da reqid 1 mode tunnel
src 0.0.0.0/0 dst 192.168.8.0/24 
	dir fwd priority 387711 
	tmpl src 139.59.60.111 dst 192.168.1.75
		proto esp reqid 1 mode tunnel
src 0.0.0.0/0 dst 192.168.8.0/24 
	dir in priority 387711 
	tmpl src 139.59.60.111 dst 192.168.1.75
		proto esp reqid 1 mode tunnel
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket in priority 0 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket out priority 0 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket in priority 0 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket out priority 0 
src ::/0 dst ::/0 
	socket in priority 0 
src ::/0 dst ::/0 
	socket out priority 0 
src ::/0 dst ::/0 
	socket in priority 0 
src ::/0 dst ::/0 
	socket out priority 0 
`

**Comment by tobiasbrunner** [comment]
If you have tunnel with `0.0.0.0/0` all traffic ends up there. You might need [bypass policies](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies).

**Reply by shripad-nesecure** [reply]
If remote TS is 0.0.0.0/0 then traffic should go via ipsec tunnel, but that should be only applicable for local_ts ,in my case which is 192.168.8.0/24, but traffic is getting affected for 192.168.1.75(wan ip) <--> 8.8.8.8,1.1.1.1 , which are not part of the Traffic selectors, and as well this behaviour is there only when there are more than 1 wan links on the device

**Reply by tobiasbrunner** [reply]
Are you sure that's the source IP that's selected?

**Reply by shripad-nesecure** [reply]
Yes, while doing tcpdump on wan, I can see packets going to 8.8.8.8 with source as 192.168.1.75, but I'm not receiving any reply, if I stop ipsec I need to restart the wan interface or router to get wan interface working again

**Reply by tobiasbrunner** [reply]
Any NAT? Or other firewall rules?

**Reply by shripad-nesecure** [reply]
No
